opencti-platform/opencti-graphql/src/manager/garbageCollectionManager.ts

```diff
@@ -0,0 +1,55 @@
+import { type ManagerDefinition, registerManager } from './managerModule';
+import conf, { ENABLED_GARBAGE_COLLECTION_MANAGER, logApp } from '../config/conf';
+import { GARBAGE_COLLECTION_MANAGER_USER, executionContext } from '../utils/access';
+import { completeDelete, findOldDeleteOperations } from '../modules/deleteOperation/deleteOperation-domain';
+
+const GARBAGE_COLLECTION_MANAGER_ENABLED = ENABLED_GARBAGE_COLLECTION_MANAGER;
+const GARBAGE_COLLECTION_MANAGER_KEY = conf.get('garbage_collection_manager:lock_key') || 'garbage_collection_manager_lock';
+const SCHEDULE_TIME = conf.get('garbage_collection_manager:interval') || 60000; // 1 minute
+const BATCH_SIZE = conf.get('garbage_collection_manager:batch_size') || 10000;
+const DELETED_RETENTION_DAYS = conf.get('garbage_collection_manager:deleted_retention_days') || 7;
+
+/**
+ * Search for N (batch_size) older than DELETED_RENTENTION_DAYS DeleteOpertions
+ * Completely delete these deleteOperations
+ */
+export const garbageCollectionHandler = async () => {
+  const context = executionContext('garbage_collection_manager');
+  const deletedRentionDaysToConsider = Math.max(DELETED_RETENTION_DAYS, 1);
+  const deleteOperationsToManage = await findOldDeleteOperations(context, GARBAGE_COLLECTION_MANAGER_USER, deletedRentionDaysToConsider, BATCH_SIZE);
+  let errorCount = 0;
+  for (let i = 0; i < deleteOperationsToManage.length; i += 1) {
+    try {
+      const deleteOperation = deleteOperationsToManage[i];
+      await completeDelete(context, GARBAGE_COLLECTION_MANAGER_USER, deleteOperation.id);
```

Comment:
In completeDelete you manage  await deleteAllObjectFiles.
Curious how do you manage files part of "file indexing feature".
What occurs when the user put the entity in the bin? Data indexed from files inside this entity are still available for search?

Reply:
We introduced a flag "removed" turned to true when item goes in the bin. Removed files are excluded from search.
We turn off the flag on restore or delete properly the file on permanent deletion.

Reply:
The data already indexed from an entity in the bin is marked as "unavailable" and doesn't come out on search. When the entity is deleted forever, the indexation is cleaned.

Reply:
Speaking under the team control, but that's what I understood :)